packages/apollo-language-server/src/schema/providers/index.ts

```diff
@@ -24,6 +24,10 @@ export function schemaProviderFromConfig(
   config: ApolloConfig
 ): GraphQLSchemaProvider {
   if (isServiceConfig(config)) {
+    if (config.service.name) {

```

Comment:
I think we should hit this branch last since you may have a name and things like a local schema file. The name is arbitrary right now (in that we want to use it for debugging, client name, etc) and we need to clean up the "engine linking" side of things.